Pal 666 service by name only

nw1984 opened the pull request.

Commit:
PAL-324-change-pull-image-policy

Commit:
Merge branch 'develop' of github.com:gchq/Palisade-services into develop

Commit:
Merge branch 'develop' of github.com:gchq/Palisade-services into develop

Commit:
Merge branch 'develop' of github.com:gchq/Palisade-services into develop

Commit:
Merge branch 'develop' of github.com:gchq/Palisade-services into develop

Commit:
Merge branch 'develop' of github.com:gchq/Palisade-services into develop

Commit:
Merge branch 'develop' of github.com:gchq/Palisade-services into develop

Commit:
PAL-666 first pass

ac74475 commented on resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java:
```diff
@@ -71,11 +69,12 @@ private LeafResource asFileResource(final File file) {
         }
-        URI dataServiceUri = clientConfiguration.getClientUri("data-service").orElseThrow(() -> new RuntimeException("Failed to find any instance of 'data-service'"));
+
+        String dataServiceName = "data-service";
```
Use the static constant above instead

dev-958 commented on resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/config/ApplicationConfiguration.java:
```diff
@@ -17,11 +17,10 @@
 
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.netflix.discovery.EurekaClient;
+import org.apache.kerby.kerberos.kerb.client.preauth.pkinit.ClientConfiguration;
```
!!! You what?

dev-958 commented on resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java:
```diff
@@ -39,10 +38,9 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(SimpleResourceService.class);
 
-    private final ClientConfiguration clientConfiguration;
+    private static final String DATA_SERVICE_NAME = "data-service";
```
Nope. This needs to be grabbed from a config file with the values duplicated in both k8s config and the eureka config, it's a reference to a data service that will handle a defined set of technologies/data servers so we may have a list of semantic handles for them.

Commit:
PAL-666 updated as a result of code review

Commit:
PAL-666 update as a result of code review comments

developer01189998819991197253 commented on resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/config/ApplicationConfiguration.java:
```diff
@@ -78,49 +76,25 @@
     /**
      * A container for a number of {@link StdResourcePrepopulationFactory} builders used for creating {@link uk.gov.gchq.palisade.resource.Resource}s
-     * These resources will be used for prepopulating the {@link ResourceService}
+     * These resources will be u fsed for prepopulating the {@link ResourceService}
```
```suggestion
     * These resources will be used for prepopulating the {@link ResourceService}
```

developer01189998819991197253 commented on resource-service/src/main/resources/application.yaml:
```diff
@@ -44,3 +44,4 @@ population:
 # Available options: simple, hadoop
 resource:
   implementation: simple
+
```
web.client.data-service: localhost:8082 missing from default application.yaml

developer01189998819991197253 replied:
added

developer01189998819991197253 committed:
updated application.yaml in resource service with web.client.data-service
updated resource service ApplicationConfiguration to fix spelling error and move @value closer to SimpleResourceService Bean

developer01189998819991197253 committed:
resolved some sonarQube code smells

developer01189998819991197253 committed:
updated some javadoc in resource service

developer01189998819991197253 committed:
added javadoc to SimpleResourceService to fix SQ code smell

ac74475 commented on resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/service/SimpleResourceService.java:
```diff
@@ -77,6 +85,13 @@ private LeafResource asFileResource(final File file) {
+    /**
+     * Query stream.
+     *
+     * @param uri  the uri
+     * @param pred the pred
+     * @return the stream
```
can you be a bit more descriptive on these

developer01189998819991197253 replied:
updated description

developer01189998819991197253 committed:
made comment a bit more descriptive in SimpleResourceServiceÂ
